darts/models/forecasting/pl_forecasting_module.py

```diff
@@ -230,6 +257,45 @@ def _compute_loss(self, output, target):
             # last dimension of model output, for properly computing the loss.
             return self.criterion(output.squeeze(dim=-1), target)
 
+    def _setup_metrics(self):
+        if self.metrics_str:
+            self.metrics = []
+            task_module = torchmetrics.functional
+            for metric in self.metrics_str:
+                try:
+                    self.metrics.append(getattr(task_module, metric))
+                except AttributeError as e:
+                    raise_log(
+                        ValueError(
+                            f"{metric} is not a valid functional metric defined in the torchmetrics.functional module"
+                        )
+                    )
+                    raise e
+
+    def calculate_metrics(self, y, y_hat, tag):
+        metrics = []
+        for metric, metric_str, metric_params in zip(
+            self.metrics, self.metrics_str, self.metrics_params
+        ):
+            if self.likelihood:
+                _metric = metric(y_hat, self.likelihood.sample(y), **metric_params)
+            else:
+                # If there's no likelihood, nr_params=1 and we need to squeeze out the
+                # last dimension of model output, for properly computing the metric.
+                _metric = metric(y_hat, y.squeeze(dim=-1), **metric_params)
+
+            metrics.append(_metric)
+
+            self.log(
+                f"{tag}_{metric_str}",
+                _metric,
+                on_epoch=True,
+                on_step=False,
+                logger=True,
+                prog_bar=True,
+            )
+        return metrics
```

Comment:
Why do we need to maintain and return `metrics` in this function? It seems it's never used.